examples/16_btd_write_parallel.cpp

```diff
@@ -0,0 +1,400 @@
+            auto lookup = extensions.find(pair.first);
+            if (lookup != extensions.end())
+            {
+                std::string extension = lookup->second;
+                res.push_back(std::move(extension));
+            }
+        }
+    }
+    return res;
+}
+
+void setupMeshComp(
+    openPMD::Container<openPMD::Mesh> &meshes, int currRank, const Workload &w)
+{
+    for (const auto &ff : m_common_fields)
+    {
+        for (const auto &cc : m_common_comps)
+        {
+            auto mesh_field = meshes[ff];
+            auto curr_mesh_comp = meshes[ff][cc];
+            Datatype datatype = determineDatatype<double>();
+            Extent global_extent = {
+                m_blockX * m_buffers.size(), m_blockY, m_blockZ};
+            Dataset dataset = Dataset(datatype, global_extent);
+
+            curr_mesh_comp.resetDataset(dataset);
+            if (m_verbose and !currRank)
+            {
+                cout << "AllRanks: Prepared a Dataset [" << ff << "/" << cc
+                     << "] of size " << dataset.extent[0] << " x "
+                     << dataset.extent[1] << " x " << dataset.extent[2]
+                     << " and Datatype " << dataset.dtype
+                     << " iteration=" << w.whichSnapshot << '\n';
+            }
+        }
+    }
+}
+
+void doFlush(
+    const Workload &w,
+    const std::unique_ptr<openPMD::Series> &series,
+    int currRank)
+{
+
+    if (m_barrier_at_flush)
+    {
+        if (m_verbose and !currRank)
+            std::cout << "AllRanks: Barrier at doFlush() ";
+
+        MPI_Barrier(MPI_COMM_WORLD);
+    }
+
+    else
+    {
+        if (m_verbose and !currRank)
+            std::cout << "AllRanks: At doFlush(), ";
+    }
+
+    if ((m_diskFlushFrequency > 0) &&
+        (w.whichBuffer % m_diskFlushFrequency) == 0)
+    {
+        if (m_verbose and !currRank)
+            std::cout << " To disk!" << std::endl;
+
+        if (m_adiosFlattenSteps)
+        {
+            series->iterations[w.whichSnapshot].seriesFlush(
+                R"(adios2.engine.preferred_flush_target = "new_step")");
+        }
+        else
+        {
+            series->iterations[w.whichSnapshot].seriesFlush(
+                R"(adios2.engine.preferred_flush_target = "disk")");
+        }
+    }
+    else
+    {
+        if (m_verbose and !currRank)
+            std::cout << " To  buffer." << std::endl;
+
+        series->iterations[w.whichSnapshot].seriesFlush(
+            "adios2.engine.preferred_flush_target = \"buffer\"");
+    }
+}
+
+void doWork(
+    const Workload &w,
+    const std::unique_ptr<openPMD::Series> &series,
+    int currRank,
+    std::string const &field_name,
+    std::string const &comp_name,
+    double seed)
+{
+
+    bool const first_write_to_iteration =
+        !series->iterations.contains(w.whichSnapshot);
+
+    auto meshes = series->iterations[w.whichSnapshot].meshes;
+
+    // is this the trouble maker?
+    series->iterations[w.whichSnapshot].open();
```

Comment:
@franzpoeschel should we use
```suggestion
    series->snapshots()[w.whichSnapshot].open();
```
now?

Reply:
Ideally, yes

Reply:
for writing, more straightforward to use iterations[]?